src/files/XmlFile.spec.ts

```diff
@@ -595,6 +595,58 @@ describe('XmlFile', () => {
     });
 
     describe('typedef', () => {
+        it('loads d.bs files from parent scope', async () => {
+            await program.addOrReplaceFile<XmlFile>('components/ParentComponent.xml', `
+                <?xml version="1.0" encoding="utf-8" ?>
+                <component name="ParentComponent" extends="Scene">
+                    <script uri="ParentComponent.brs" />
+                </component>
+            `);
+
+            await program.addOrReplaceFile('components/ParentComponent.d.bs', `
+                import "Lib.brs"
+                namespace Parent
+                    sub log()
+                    end sub
+                end namespace
+            `);
+            await program.addOrReplaceFile('components/ParentComponent.brs', `
+                sub Parent_log()
+                end sub
+            `);
+
+            await program.addOrReplaceFile('components/Lib.d.bs', `
+                namespace Lib
+                    sub log()
+                    end sub
+                end namespace
+            `);
+            await program.addOrReplaceFile('components/Lib.brs', `
+                sub Lib_log()
+                end sub
+            `);
+
+            await program.addOrReplaceFile<XmlFile>('components/ChildComponent.xml', `
+                <?xml version="1.0" encoding="utf-8" ?>
+                <component name="ChildComponent" extends="ParentComponent">
+                    <script uri="ChildComponent.bs" />
+                </component>
+            `);
+            await program.addOrReplaceFile('components/ChildComponent.bs', `
+                sub init()
+                    Parent.log()
+                    Lib.log()
+                end sub
+            `);
+            await program.validate();
+            expect(program.getDiagnostics()[0]?.message).not.to.exist;
```

Comment:
This test always looks weird. You should be testing that the collection is empty.

Reply:
The reason I do it this way is because `"expected array of length 1 to be length 0"` isn't overly helpful when trying to diagnose why a test is failing. This test does _technically_ check if the collection is empty. This pattern will print the first diagnostic to the console when the test fails, which is nice to see the problem right away in the console, as well as in Github Actions because we can't debug unit tests there.

Reply:
Oh ok that's a fair reason!